File: OOP C++/lab2/Less.cpp
#include "Less.h"
#include "FactoryInitializer.h"


void Less::Execute(Operands& operands, Tokens& tokens, std::string& output, Reader& reader) {
    int operand1 = operands.GetAndPop();
    int operand2 = operands.GetAndPop();
    operands.Push(operand2 < operand1);

}

namespace {
    FactoryInitializer<Less> Registration("<");
}

File: Lab1/src/Commands/Less.java
package Commands;

import Utility.Operands;

import java.util.StringTokenizer;

public class Less implements Command{
    @Override
    public void Execute(StringTokenizer tokens, Operands operands) {
        int num1 = operands.GetAndPop();
        int num2 = operands.GetAndPop();
        if (num2 < num1) {
            operands.Push(1);
        } else {
            operands.Push(0);
        }
    }

    @Override
    public void Check(StringTokenizer tokens, Operands operands) {

    }

}
